scheduler.go

```diff
@@ -1335,3 +1341,19 @@ func (s *Scheduler) StopBlockingChan() {
 	}
 	s.startBlockingStopChanMutex.Unlock()
 }
+
+// EnableTelemetry enables the collection of metrics for the scheduler. Takes
+// the name of the scheduler to set it as a metric label
+func (s *Scheduler) EnableTelemetry(schedulerName string) {
+	initTelemetry()
+	s.isTelemetryEnabled = true
+	s.telemetryShdName = schedulerName
+}
+
+func (s *Scheduler) observeJobLatency(jobName string, latency float64) {
```

Comment:
rather than have this method on the scheduler, how about if we make a separate `telemetry` struct that both the scheduler and executor can take pointers to. Then we don't need the `isTelemetryEnabled` bool because the struct will either exist or be nil.

Right now, it feels odd to me that the executor is taking in a scheduler and then calling a metrics method on the scheduler.